src/software/ai/hl/stp/tactic/crease_defender/simulated_crease_defender_tactic_test.cpp

```diff
@@ -0,0 +1,132 @@
+    setBallState(BallState(Point(4.5, -3), Vector(0, 0)));
+    addFriendlyRobots(TestUtil::createStationaryRobotStatesWithId({initial_position}));
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId(
+        {Point(1, 0), enemy_threat_point, Point(1, -1.5), field().enemyGoalCenter(),
+         field().enemyDefenseArea().negXNegYCorner(),
+         field().enemyDefenseArea().negXPosYCorner()}));
+
+    auto robot_navigation_obstacle_config =
+        std::make_shared<RobotNavigationObstacleConfig>();
+
+    auto tactic =
+        std::make_shared<CreaseDefenderTactic>(robot_navigation_obstacle_config);
+    tactic->updateControlParams(enemy_threat_point, alignment);
+    setTactic(tactic);
+    setRobotId(0);
+    setMotionConstraints({MotionConstraint::ENEMY_ROBOTS_COLLISION,
+                          MotionConstraint::FRIENDLY_DEFENSE_AREA});
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [target_defend_location, tactic](std::shared_ptr<World> world_ptr,
+                                         ValidationCoroutine::push_type& yield) {
+            // Check that conditions hold for 1000 ticks
+            unsigned num_ticks = 1000;
+            for (unsigned i = 0; i < num_ticks; i++)
+            {
+                robotInPolygon(0, target_defend_location, world_ptr, yield);
+            }
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(10));
+}
+
+INSTANTIATE_TEST_CASE_P(
+    CreaseDefenderEnvironment, SimulatedCreaseDefenderTacticPositionTest,
+    ::testing::Values(
+        // Enemy threat in front of crease, LEFT
+        std::make_tuple(Point(1, 2.5), CreaseDefenderAlignment::LEFT,
+                        Rectangle(Point(-3.5, 1), Point(-3.25, 0.8))),
+        // Enemy threat in front of crease, CENTRE
+        std::make_tuple(Point(1, -2.5), CreaseDefenderAlignment::CENTRE,
+                        Rectangle(Point(-3.5, -0.5), Point(-3.25, -0.7))),
+        // Enemy threat in front of crease, RIGHT
+        std::make_tuple(Point(1.5, 2), CreaseDefenderAlignment::RIGHT,
+                        Rectangle(Point(-3.25, 0.7), Point(-3, 0.5))),
+        // Enemy threat left side of crease, RIGHT
+        std::make_tuple(Point(-4.25, 2.5), CreaseDefenderAlignment::RIGHT,
+                        Rectangle(Point(-4.3, 1), Point(-4.5, 1.2))),
+        // Enemy threat left side of crease, CENTRE
+        std::make_tuple(Point(-4, 2.5), CreaseDefenderAlignment::CENTRE,
+                        Rectangle(Point(-4.2, 1), Point(-4.3, 1.2))),
+        // Enemy threat right side of crease, RIGHT
+        std::make_tuple(Point(-4, -2), CreaseDefenderAlignment::RIGHT,
+                        Rectangle(Point(-4.3, -1), Point(-4.1, -1.2))),
+        // Enemy threat right side of crease, LEFT
```

Comment:
I don't see an exact pattern with the Points and Rectangles created here, were the values determined experimentally?

Reply:
Yeah, they're too specific. I changed it to make the regions larger and more intuitive